hydroflow/examples/chat/helpers.rs

```diff
@@ -52,3 +52,17 @@ pub fn connect_get_addr(m: Message) -> Option<SocketAddr> {
         _ => None,
     }
 }
+
+pub fn resolve_ipv4_connection_addr(server_ip: String, server_port: u16) -> Option<SocketAddr> {
+    let addrs = format!("{}:{}", server_ip, server_port)
+        .to_socket_addrs()
+        .unwrap();
+
+    for addr in addrs {
+        if addr.is_ipv4() {
+            return Some(addr);
+        }
+    }
+
+    return None;
+}
```

Comment:
Is there a reason instead of using just `format!("{}:{}", server_ip, server_port).parse()` like before?

Reply:
So if `server_ip` is a DNS name rather than an IP address, it's possible for it to resolve to both an IPv6 and an IPv4 version; it wasn't clear to me from the documentation how to control whether the IPv6 or IPv4 address was selected in that case. Also, honestly, I found an example of `to_socket_addrs()` and figured "hey, this'll work".